plugins/algoltek-usb/fu-algoltek-usb-device.c

```diff
@@ -0,0 +1,633 @@
+/*
+ * Copyright (C) 2023 Ling.Chen <ling.chen@algoltek.com.tw>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-algoltek-usb-common.h"
+#include "fu-algoltek-usb-device.h"
+#include "fu-algoltek-usb-firmware.h"
+// #include "fu-algoltek-usb-struct.h"
+
+struct _FuAlgoltekUsbDevice {
+	FuUsbDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuAlgoltekUsbDevice, fu_algoltek_usb_device, FU_TYPE_USB_DEVICE)
+
+static void
+fu_algoltek_usb_device_to_string(FuDevice *device, guint idt, GString *str)
+{
+	FU_DEVICE_CLASS(fu_algoltek_usb_device_parent_class)->to_string(device, idt, str);
+}
+
+static guint8
+checkSum(GByteArray *byteArray)
+{
+	guint8 result = 0;
+	for (guint32 i = 0; i < byteArray->len; i++) {
+		result += (guint8)(byteArray->data[i] & 0xff);
+	}
+	result = (guint8)(~result + 1);
+	return result;
```

Comment:
I think this could be all be replaced by `~fu_sum8(byteArray->data, byteArray->len) + 1`

Reply:
All check_sum have been replaced by `~fu_sum8`.